src/pages/DatasetRegistration.js

```diff
@@ -119,65 +96,16 @@ class DatasetRegistration extends Component {
   }
 
   async init() {
-    let formData = Storage.getData('dar_application') === null ? this.state.formData : Storage.getData('dar_application');
-    Storage.removeData('dar_application');
-    if (this.props.location.props !== undefined && this.props.location.props.formData !== undefined) {
-      if (this.props.location.props.formData.darCode !== undefined) {
-        formData = this.props.location.props.formData;
-        formData.ontologies = this.getOntologies(formData);
-      } else if (this.props.location.props.formData.datasetIds !== undefined) {
-        // set datasets sent by data set catalog
-        formData.datasets = this.processDataSet(this.props.location.props.formData.datasetIds);
-      }
-    }
+    let formData = Storage.getData('dataset_registration') === null ? this.state.formData : Storage.getData('dataset_registration');
+    Storage.removeData('dataset_registration');
```

Comment:
We use storage to stash content into a cookie in case we go back and need to re-read it for some reason. I don't think we should be using it very much, and in fact, I'd like to remove it if at all possible. I would try and not use it here.